### PR TITLE
Dagna animations

### LINKED ISSUE #749: [FEATURE] D'Agna model animations
## 🚀 Feature

Add D'Agna model animations in its animator.

### Animations
https://drive.google.com/file/d/19ZtDa4zp_xT54471VAsmhvAAJiKJ9Jh_/view?usp=drive_link

### PR DESCRIPTION
**Pull request by BertovDev**
This PR implements Dagna animations for Dagna.

Please test the PR and leave feedbacks for the animation tempo.

To test:

1. Scherzo its just the same as walking animation but slowed
2. Skill1 Dance Macabre , its a little dance in the place
3. Skill2 Pizzicato , a tittle shotgun in the place
4. Skill3 Band recluit , a tittle jump with a tittle movement ( should i remove the movement )
5. Skill4 Orchesta, no animation

closes #749

### CODE REVIEW

**Comment by mgrunwaldt** [comment]
If I click on D'Agna (quick play) no confirm button appears.
image

See for other players how it looks 
image


I'm not testing locally, I'm testing with brazil, could this be it?

**Comment by mgrunwaldt** [comment]
Great job, a few comments:
- Scherzo is a bit laggy (see video 1)
- If I do skill 3 (jump) and really quickly move the animation is cancelled but not the cooldown of the skill, this feels weird (video 2)
- For skill 2 (shotgun) i feel that the end is very abrupt, like there should be an animation from that pose to the natural pose (video 3). Is this something you can achieve in Unity or do I need to talk to the 3d modeler?
- Once or twice animations freezed (video 4)

https://github.com/lambdaclass/curse_of_myrra/assets/22064054/794821a7-e3f6-49a4-84dc-96a1d3c70472


https://github.com/lambdaclass/curse_of_myrra/assets/22064054/76b2cb19-af31-48a5-b4de-9dc59cd2503d




https://github.com/lambdaclass/curse_of_myrra/assets/22064054/aeb54585-90bf-4281-89d7-8cb3d72db196



https://github.com/lambdaclass/curse_of_myrra/assets/22064054/5cd9f501-9222-4d02-86c8-4f2abc38dc15

**Comment by BertovDev** [comment]
> Great job, a few comments:
> 
>     * Scherzo is a bit laggy (see video 1)
> 
>     * If I do skill 3 (jump) and really quickly move the animation is cancelled but not the cooldown of the skill, this feels weird (video 2)
> 
>     * For skill 2 (shotgun) i feel that the end is very abrupt, like there should be an animation from that pose to the natural pose (video 3). Is this something you can achieve in Unity or do I need to talk to the 3d modeler?
> 
>     * Once or twice animations freezed (video 4)
> 
> 
> Grabacion.de.pantalla.2023-08-08.a.la.s.16.38.19.mov
> Grabacion.de.pantalla.2023-08-08.a.la.s.16.45.25.mov
> Grabacion.de.pantalla.2023-08-08.a.la.s.16.47.17.mov
> Grabacion.de.pantalla.2023-08-08.a.la.s.16.46.54.mov

This Feedback is awesome! Thank you so much.
Scherzo dont have animation its just the walking slowed, its true that looks a little laggy. I will try to debug it.
For skill3 and skill2 i will fix it! nice catch
And finally i cant reproduce the bug in video 4

File: client/Assets/Scripts/Battle.cs [hunk 1]
```diff
@@ -447,6 +447,17 @@ private void updatePlayer(GameObject player, Player playerUpdate, long pastTime)
...
+            .GetComponent<Character>()
+            .CharacterModel.GetComponent<Animator>();
+        // if (playerUpdate.Effects.ContainsKey((ulong)PlayerEffect.Poisoned))
+        // {
+        //     GetComponent<PlayerFeedbacks>().SetActivePoisonedFeedback(player, true);
+        // }
+        // else
+        // {
+        //     GetComponent<PlayerFeedbacks>().SetActivePoisonedFeedback(player, false);
+        // }
```
Comment (mgrunwaldt) [inline client/Assets/Scripts/Battle.cs:460]: Why is this left as a comment? Can't we just delete it?

File: client/Assets/Scripts/Battle.cs [hunk 2]
```diff
@@ -831,6 +842,11 @@ float characterSpeed
             }
         }
 
+        // if (playerUpdate.Effects.ContainsKey((ulong)PlayerEffect.Scherzo))
+        // {
+        //     characterSpeed *= 0.5f;
+        // }
```
Comment (mgrunwaldt) [inline client/Assets/Scripts/Battle.cs:848]: Same as previous comment

File: client/Assets/Scripts/Battle.cs [hunk 3]
```diff
@@ -870,6 +886,7 @@ float characterSpeed
         if (playerUpdate.Effects.ContainsKey((ulong)PlayerEffect.Slowed))
         {
             characterSpeed *= 0.5f;
+            print("slowed");
```
Comment (mgrunwaldt) [inline client/Assets/Scripts/Battle.cs:889]: Delete the print before merging to main

**Review by mgrunwaldt — COMMENTED** [review]
Works good but there are a few unnecesary comments in the code

**Review by mgrunwaldt — APPROVED** [review]
Looks good!

File: client/Assets/Scripts/Battle.cs [hunk 4]
```diff
@@ -447,6 +447,9 @@ private void updatePlayer(GameObject player, Player playerUpdate, long pastTime)
         */
         Character character = player.GetComponent<Character>();
         var characterSpeed = PlayerControls.getBackendCharacterSpeed(playerUpdate.Id) / 100f;
+        Animator mAnimator = player
```
Comment (AngieDutra) [inline client/Assets/Scripts/Battle.cs:450]: just a minor change: `mAnimator`? I would go for `modelAnimator` to make this clearer

**Review by AngieDutra — APPROVED** [review]
LGTM!

**Review by ncontinanza — APPROVED** [review]
Nice job!

**Pull request merged**